pypeit/spectrographs/subaru_focas.py

```diff
@@ -0,0 +1,688 @@
+        # TODO
+        # nimg = 1
+        # mosaic = None
+        detector = self.get_detector_par(det, hdu=hdu_l)
+
+        # Need the exposure time
+        exptime = float(head['EXPTIME'])
+        bin_x = int(head['BIN-FCT1'])
+
+        arr_shape = data.shape
+        # allocate output arrays
+        rawdata = data.astype(float)
+        rawdatasec_img = np.zeros(arr_shape, dtype=int)
+        oscansec_img = np.zeros(arr_shape, dtype=int)
+
+        # collect correct data & overscan for binning in spatial (X) axis
+        oscan_arr = overscan[bin_x][(det - 1)*4:][:4]
+        assert len(oscan_arr) == 4
+
+        # fill in rawdatasec_img and oscansec_img arrays according to
+        # specification above
+        for i, oscan in enumerate(oscan_arr):
+            n_amp = i + 1
+
+            if n_amp in [2, 4]:
+                lft_oscan_start, lft_oscan_end = oscan[0] - 1, oscan[1]
+                oscansec_img[:, lft_oscan_start:lft_oscan_end] = n_amp
+
+            data_start, data_end = oscan[2] - 1, oscan[3]
+            rawdatasec_img[:, data_start:data_end] = n_amp
+
+            if n_amp in [1, 3]:
+                rgt_oscan_start, rgt_oscan_end = oscan[4] - 1, oscan[5]
+                oscansec_img[:, rgt_oscan_start:rgt_oscan_end] = n_amp
+
+        return (detector, rawdata, hdu_l, exptime, rawdatasec_img, oscansec_img)
+
+
+    @property
+    def allowed_mosaics(self):
+        """
+        Return the list of allowed detector mosaics.
+
+        Subaru FOCAS only allows for mosaicing both detectors.
+
+        Returns:
+            :obj:`list`: List of tuples, where each tuple provides the 1-indexed
+            detector numbers that can be combined into a mosaic and processed by
+            PypeIt.
+        """
+        return [(1,2)]
+
+    @property
+    def default_mosaic(self):
+        return self.allowed_mosaics[0]
+
+
+    def raw_header_cards(self):
+        """
+        Return additional raw header cards to be propagated in
+        downstream output files for configuration identification.
+
+        The list of raw data FITS keywords should be those used to populate
+        the :meth:`~pypeit.spectrographs.spectrograph.Spectrograph.configuration_keys`
+        or are used in :meth:`~pypeit.spectrographs.spectrograph.Spectrograph.config_specific_par`
+        for a particular spectrograph, if different from the name of the
+        PypeIt metadata keyword.
+
+        This list is used by :meth:`~pypeit.spectrographs.spectrograph.Spectrograph.subheader_for_spec`
+        to include additional FITS keywords in downstream output files.
+
+        Returns:
+            :obj:`list`: List of keywords from the raw data files that should
+            be propagated in output files.
+        """
+        return ['DISPERSR', 'FILTER02', 'SLIT', 'SLT-LEN', 'SLT-PA',
+                'SLT-WID']  #, 'SLTCPIX1', 'SLTCPIX2', 'SLTC-RA', 'SLTC-DEC']
+
+
+# Definitions for the over scan regions in the DS9 image coordinate.
```

Comment:
are these used somewhere?  if so, wrap them in a `def`
If not, please remove

Reply:
Wrapped!